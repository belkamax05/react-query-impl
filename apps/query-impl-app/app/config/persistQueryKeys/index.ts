import { QueryKeyJoin } from '@query-impl/core';
import { Call, Tuples } from 'hotscript';
import queryKeys from '../queryKeys';
/**
 * list of query keys to be saved into localStorage and restored on each session
 */
const persistQueryKeys = [queryKeys.samples.counter, queryKeys.samples.todo];

export type PersistQueryKeys = typeof persistQueryKeys;

export type PersistQueryKeysUnion = Call<Tuples.ToUnion, PersistQueryKeys>;

export type PersistQueryKeysKey = QueryKeyJoin<PersistQueryKeysUnion>;

export default persistQueryKeys;
