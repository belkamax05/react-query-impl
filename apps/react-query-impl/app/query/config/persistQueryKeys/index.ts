import queryKeys from '../queryKeys';

/**
 * list of query keys to be saved into localStorage and restored on each session
 */
const persistQueryKeys = [queryKeys.todo] as const;

export default persistQueryKeys;
