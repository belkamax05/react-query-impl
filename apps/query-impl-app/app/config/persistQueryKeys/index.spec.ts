import persistQueryKeys from '.';

describe('persistQueryKeys', () => {
  it('should match snapshot', () => {
    expect(persistQueryKeys).toMatchSnapshot();
  });
});
