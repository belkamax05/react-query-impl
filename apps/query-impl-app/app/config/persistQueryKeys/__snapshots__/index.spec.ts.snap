// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`persistQueryKeys should match snapshot 1`] = `
[
  [
    "samples",
    "counter",
  ],
  [
    "samples",
    "todo",
  ],
]
`;
